contracts/fair-burn/src/state.rs

```diff
@@ -0,0 +1,29 @@
+use crate::ContractError;
+
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{ensure, Decimal, Storage};
+use cw_storage_plus::Item;
+
+#[cw_serde]
+pub struct Config {
+    /// The percentage of funds to be burned
+    pub fee_percent: Decimal,
+}
+
+impl Config {
+    pub fn save(&self, storage: &mut dyn Storage) -> Result<(), ContractError> {
+        self.validate()?;
+        CONFIG.save(storage, self)?;
+        Ok(())
+    }
+
+    fn validate(&self) -> Result<(), ContractError> {
+        ensure!(
+            self.fee_percent > Decimal::zero(),
```

Comment:
`Decimal` is a `Uint128` internally, so it's already unsigned, so this check may not be needed.

Reply:
this just checks to make sure it's greater than zero, to make sure something gets burned